hack/gen_rvar_table.py

```diff
@@ -0,0 +1,119 @@
+#!/bin/python
+
+import re
+
+with open('defaults/main.yml') as f:
+    lines = f.read().split("\n")
+
+def print_rvar_table(rvar_table):
+    keys = ["name", "description", "default", "required"]
+
+    sizes = {}
+    for key in keys:
+        sizes[key] = len(key)
+
+    for item in rvar_table:
+        for key in item.keys():
+            sizes[key] = max(sizes[key], len(item[key]))
+
+    headers = []
+    delimiters = []
+    for key in keys:
+        headers.append(" {}{} ".format(key, " "*(sizes[key]-len(key))))
+        delimiters.append("-{}-".format("-"*(sizes[key])))
+    print "|{}|".format("|".join(headers))
+    print "|{}|".format("|".join(delimiters))
+
+    # sort items by variable name
+    ki_items = {}
+    for item in rvar_table:
+        ki_items[item["name"]] = item
+
+    for key in sorted(ki_items):
+        item = ki_items[key]
+        data = []
+        for key in keys:
+            data.append(" {}{} ".format(item[key], " "*(sizes[key]-len(item[key]))))
+        print "|{}|".format("|".join(data))
+
+def get_delims(line, skip_list=False):
+    # replace all "...", '...' and [...] with # so it is easy to determine
+    # position of the ',' symbol
+    line = re.sub(r'("[^"]*")', lambda l: '#'*len(l.group()) , line)
+    line = re.sub(r"('[^']*')", lambda l: '#'*len(l.group()) , line)
+    if not skip_list:
+        line = re.sub(r"(\[[^\]]*\])", lambda l: '#'*len(l.group()) , line)
+    return re.findall(r"[^,]+", line)
+
+def parse_list(value):
```

Comment:
That already crossed my mind. I rejected that cause I did not like the syntax. You have to specify additional quotes and other stuff. But after we discuss advantages and disadvantages of this approach and conclude it is still way to go, let's do it.